packages/mdc-switch/foundation.ts

```diff
@@ -21,47 +21,51 @@
  * THE SOFTWARE.
  */
 
-import MDCFoundation from '@material/base/foundation';
-import MDCSwitchAdapter from './adapter';
-/* eslint-enable no-unused-vars */
+import {MDCFoundation} from '@material/base/foundation';
+import {MDCSwitchAdapter} from './adapter';
 import {cssClasses, strings} from './constants';
 
 /**
- * @extends {MDCFoundation<!MDCSwitchAdapter>}
+ * Foundation for the MDC Switch. Encapsulates business logic for the switch.
+ *
+ * See architecture documentation for more details.
+ * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
  */
-class MDCSwitchFoundation extends MDCFoundation {
-  /** @return enum {string} */
+class MDCSwitchFoundation extends MDCFoundation<MDCSwitchAdapter> {
+  /** The string constants used by the switch. */
   static get strings() {
     return strings;
   }
 
-  /** @return enum {string} */
+  /** The CSS classes used by the switch. */
   static get cssClasses() {
     return cssClasses;
   }
 
-  /** @return {!MDCSwitchAdapter} */
-  static get defaultAdapter() {
-    return /** @type {!MDCSwitchAdapter} */ ({
-      addClass: (/* className: string */) => {},
-      removeClass: (/* className: string */) => {},
-      setNativeControlChecked: (/* checked: boolean */) => {},
-      setNativeControlDisabled: (/* disabled: boolean */) => {},
-    });
+  /** The default Adapter for the switch. */
+  static get defaultAdapter(): MDCSwitchAdapter {
+    return {
+      // tslint:disable:no-empty
+      addClass: () => {},
```

Comment:
I've been changing these to `() => undefined`, then removing the tslint:disables